extensions/stats/plugin.cc

```diff
@@ -561,17 +579,27 @@ bool PluginRootContext::report(::Wasm::Common::RequestInfo& request_info,
     uint64_t response_flags = 0;
     getValue({"response", "flags"}, &response_flags);
```

Comment:
@gargnupur I wonder if there is any difference between this check and just looking at `end_stream`. We just want to stop waiting to report until the end, right?

Reply:
When response flags is set.. `end_stream` would be set right? In that case, it should be fine.
We just wanted to stop waiting in error but stopping when `end_stream` is reached should also work.

Reply:
Ok, simplified to stop waiting on the end stream.